scripts/dockerlib.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+# Author Gabriel Hartmann <gabhart@microsoft.com>
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Open Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#--------------------------------------------------------------------------
+
+file_name=${0##*/}
+
+timestamp() {
+    date +"%Y-%m-%d %H:%M:%S"
```

Comment:
This is fine, probably good. I'm just always a little leery of lots of one-line functions that aren't actually used in multiple places. The argument is that it makes the code more abstract and thus more readable. It also makes the code more abstract and *less* readable.